src/components/tracked-controls.js

```diff
@@ -75,31 +89,73 @@ module.exports.Component = registerComponent('tracked-controls', {
     var deltaControllerPosition = this.deltaControllerPosition;
     var dolly = this.dolly;
     var el = this.el;
-    var orientation;
     var pose;
-    var position;
     var standingMatrix = this.standingMatrix;
     var vrDisplay = this.system.vrDisplay;
+    var hand = (controller ? controller.hand : undefined) || DEFAULT_HAND;
+    var headEl = this.getHeadElement();
+    var headObject3D = headEl.object3D;
+    var headCamera = headEl.components.camera;
+    var userHeight = (headCamera ? headCamera.data.userHeight : 0) || this.defaultUserHeight();
 
     if (!controller) { return; }
 
     // Compose pose from Gamepad.
     pose = controller.pose;
-    orientation = pose.orientation || ZERO_ORIENTATION;
-    position = pose.position || ZERO_POSITION;
-    controllerQuaternion.fromArray(orientation);
-    dolly.quaternion.fromArray(orientation);
-    dolly.position.fromArray(position);
+    // If no orientation, use camera.
+    if (pose.orientation) {
+      dolly.quaternion.fromArray(pose.orientation);
+    } else {
+      dolly.quaternion.copy(headObject3D.quaternion);
+    }
+    if (pose.position) {
+      dolly.position.fromArray(pose.position);
+    } else {
+      // The controller is not 6DOF, so apply arm model.
+      // Use controllerPosition and deltaControllerPosition to avoid creating yet more variables...
+
+      // Use camera position as head position.
```

Comment:
Can we put all this code in its own method `applyArmModel(controllerPosition)` ?